modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -226,10 +229,26 @@ const FrontendDataSet = ({
 	const isMounted = useIsMounted();
 
 	function updateDataSetItems(dataSetData) {
-		setTotal(dataSetData.totalCount);
+		setCollectionActions(dataSetData.actions);
```

Comment:
Related to https://github.com/liferay-frontend/liferay-portal/pull/3781#discussion_r1374563539, this does not look right. `actions` property in returned data contains item actions, not collection actions. Even if headless team would add collection actions to returned data, what if there are no items? There is no returned data. That's why collection actions need to be added through main API, the `creationMenu` prop.

Reply:
There is a list of actions for each item inside `dataSetData.items`. If there are no items we could have global actions.

![image](https://github.com/liferay-frontend/liferay-portal/assets/132653342/e37b0282-6b00-4c90-9876-38a66b71a74a)

Reply:
Oh, I see. Hmm, so create action can be read from two places, `creationMenu` and returned data I think `creationMenu` should have higher priority, as it represents custom implementation. Take for example what we're doing on Data Sets page, we're [opening a complex modal](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx#L823-L845). Create action taken from from returned data would be useless. I am not sure if it's even worth setting as a direct fallback, you will almost always need some inputs to populate a `POST`.

Reply:
They are two different things. `creationMenu` returns the information available in DSM Actions, actions created by the user. The actions from the returned data are part of the API call (headless). We only need the actions from the API when some DSM Action has a `permissionKey` so, we could check if it is available for the current user and we can display.

Basically, we are delaying the rendering of the plus button until we know which actions we should display. Otherwise we could start rendering the FDS with the plus button and then, when the API data finishes, removing it because DSM Actions doesn't match the `permissionKey` criteria.

Reply:
Got it! Yeah, this makes sense in principle :+1: I misread code, thinking that we're merging both sets of actions.